### ISSUE TITLE
Should we upload the unpacked files instead of gz files

### ISSUE THREAD

**Issue by jjgao** [cBioPortal/datahub#68]
For discussion:

Often times, I just wanted to look at one file, but I have to download the whole study to do that.

Gzipped files are also not good for comparison / keep tracks of changes.

And if I find a small issue, I wanted to be able to fix just one txt file instead of uploading the whole gzipped file.

**Comment by pieterlukasse** [comment]
@jjgao I agree. The only good reason for having gzipped files is because it is easy to download them (also used by the portal now when you click on the download option of a study in cBioPortal.org).

Perhaps we can move to normal git and use something like https://minhaskamal.github.io/DownGit/#/home (although this won't work for files stored in git lfs...). Ideas?

**Comment by sandertan** [comment]
Using Git LFS with uncompressed files can become quite expensive though:

> One data pack costs $5 per month, and provides a monthly quota of 50 GB for bandwidth and 50 GB for storage.


https://help.github.com/articles/about-storage-and-bandwidth-usage/#purchasing-additional-storage-and-bandwidth

**Comment by jjgao** [comment]
Wondering if normal git can support files of 100s of MB?

**Comment by pieterlukasse** [comment]
GitHub will warn you when pushing files larger than 50 MB. You will not be allowed to push files larger than 100 MB (from https://help.github.com/articles/working-with-large-files/)

**Comment by pieterlukasse** [comment]
@jjgao we will need some extra information to make a decision here:
 - how big is the total repository when we unzip all data?
 - who can write (or find) a good script to download all the files of one study? E.g. something like `./getstudy.sh -s brca_tcga` which then downloads all `brca_tcga` files directly from github. Probably github has some APIs to make this easy. Ideally, for downloading there is no dependency on git lfs at the client's side.

**Comment by jjgao** [comment]
Not sure about how big the total repo would be. @angelicaochoa is there an easy way to find out? E.g. the total size of all TCGA provisional studies would give us a rough idea.

For the 2nd one, maybe we can point users to a http server (we are currently set it up for download sql dumps) which can have both zipped and unzipped file? That way, we don't have to rely on the git bandwidth and the git repo can be for version purposes only. 

cc'ing @n1zea144 and @sheridancbio on this as well.

**Comment by jjgao** [comment]
@pieterlukasse and I had some discussion on this. We think we should start to unpack the files on datahub. @pieterlukasse will look into DownGit for downloading option.

**Comment by pieterlukasse** [comment]
Decisions (datahub call):

- [ ] @yichaoS : unpack files into their study subfolders. E.g. https://github.com/cBioPortal/datahub/blob/master/public/acc_tcga.tar.gz should become https://github.com/cBioPortal/datahub/blob/master/public/acc_tcga/ containing the unpacked staging files. Delete the .gz file.  
- [ ] @yichaoS : make a new PR
- [ ] @pieterlukasse work on download solution

**Comment by yichaoS** [comment]
PR for unzipping: https://github.com/cBioPortal/datahub/pull/152

**Comment by sandertan** [comment]
PR to update the documentation with links to datahub datasets: https://github.com/cBioPortal/cbioportal/pull/3841

**Comment by yichaoS** [comment]
Frontend PR for updating the download links: https://github.com/cBioPortal/datahub/issues/68

**Comment by pieterlukasse** [comment]
@yichaoS I couldn't find any PR in frontend on your name? What is the link?

**Comment by sandertan** [comment]
@pieterlukasse https://github.com/cBioPortal/cbioportal-frontend/pull/943
